internal/provider/kubernetes/topology_injector.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-openapi/jsonpointer"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
+)
+
+type ProxyTopologyInjector struct {
+	client.Client
+	Decoder admission.Decoder
+}
+
+func (m *ProxyTopologyInjector) Handle(ctx context.Context, req admission.Request) admission.Response {
+	binding := &corev1.Binding{}
+	if err := m.Decoder.Decode(req, binding); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if binding.Target.Name == "" {
+		return admission.Allowed("skipped")
+	}
+
+	podName := types.NamespacedName{
+		Namespace: binding.Namespace,
+		Name:      binding.Name,
+	}
+
+	pod := &corev1.Pod{}
+	if err := m.Get(ctx, podName, pod); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// Skip non-proxy pods
+	if v, ok := pod.Labels["app.kubernetes.io/component"]; !ok || v != "proxy" {
```

Comment:
can we also add some more in here ?
app.kubernetes.io/managed-by: envoy-gateway
app.kubernetes.io/name: envoy